pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -56,24 +56,32 @@ func (p *PoolManager) AllocateVirtualMachineMac(virtualMachine *kubevirt.Virtual
 		"namespace", virtualMachine.Namespace,
 		"interfaces", virtualMachine.Spec.Template.Spec.Domain.Devices.Interfaces)
 
-	for idx, iface := range virtualMachine.Spec.Template.Spec.Domain.Devices.Interfaces {
+	copyVM := virtualMachine.DeepCopy()
```

Comment:
Please comment in the commit message when the allocation could fail

Reply:
done